metrics/test/test_cognitive_complexity.py

```diff
@@ -81,13 +83,51 @@ def test_breaks_in_linear_flow_returns_3(self):
         ]
         self.assertCountEqual(locations, expected_locations)
 
-    def test_cognitive_complexity_returns_12(self):
+    def test_nesting_returns_5(self):
+        new_source_repository = SourceRepository(
+            Path(
+                "./metrics/test/data/test_cognitive_complexity/test_nested_controlflows"
+            )
+        )
+        cognitive_complexity = CognitiveComplexity()
+        cognitive_complexity._source_repository = new_source_repository
+        actual = 0
+        for (
+            file_path,
+            file_info,
+        ) in cognitive_complexity._source_repository.files.items():
+            ast: tree_sitter.Tree = cognitive_complexity._source_repository.get_ast(
+                file_info
+            )
+            tree_sitter_language = (
+                cognitive_complexity._source_repository.tree_sitter_languages[
+                    file_info.language
+                ]
+            )
+
+            initial_nodes_query_str = (
+                f"""{translate_to[file_info.language]["function"]} @func"""
+            )
+            if file_info.language == "c_sharp" or file_info.language == "java":
+                initial_nodes_query_str += (
+                    f"""{translate_to[file_info.language]["constructor"]} @cons"""
+                )
+            query_initial_nodes = tree_sitter_language.query(initial_nodes_query_str)
+            initial_nodes = query_initial_nodes.captures(ast.root_node)
+            for node, _ in initial_nodes:
+                count = cognitive_complexity._count_nesting(node, file_info)
+                actual += count
+
+        expected = 5
+        self.assertEqual(actual, expected)
+
+    def test_cognitive_complexity_returns_14(self):
         new_source_repository = SourceRepository(
             Path("./metrics/test/data/test_cognitive_complexity")
         )
         cognitive_complexity = CognitiveComplexity()
         cognitive_complexity._source_repository = new_source_repository
         result = cognitive_complexity.run()
         actual = result.outcome
-        expected = 12
+        expected = 14
```

Comment:
The 12 is not random, it is a value I wrote in #95 as a sum of the expectations for each of the individual programming languages since all three languages are now being considered in the same test where they were separate prior to #95.

I count 4 methods violating that cognitive complexity needs to be less than 5 in `test_nested_controlflows` folder:
- `metrics/test/data/test_cognitive_complexity/test_nested_controlflows/data_nesting.cs`'s `method1`
  - `switch` on line 9
  - `if` on line 14
  - `if` on line 17
  - `catch` line 19
  - `break` line 23
  - `break` line 26
- `metrics/test/data/test_cognitive_complexity/test_nested_controlflows/data_nesting.java`'s `Main`:
  - 1 for each of the `if` on lines 4-7 (for ifs total)
  - 1 for reaching 4 nesting levels
- `metrics/test/data/test_cognitive_complexity/test_nested_controlflows/data_nesting.java`'s `method1`:
  - `switch` on line 18
  - `switch` on line 21
  - `if` on line 23
  - `break` on line 26
  - `catch` on line 28
  - `break` on line 31
- `metrics/test/data/test_cognitive_complexity/test_nested_controlflows/data_nesting.py`'s `nesting1`
  - `if`on line 3
  - `for` on line 5
  - `while` on line 7
  - `if` on line 10
  - Reaching a nesting level greater than or equal to 4 on line 11
  - `for` on line 15 
  - `while` on line 15
  - `if` on line 20
  - Reaching a nesting level greater than or equal to 4 on line 21
  - `except` on line 22

Either I am missing something, or it is a good idea to label individual issues that lead to the greater violation of cognitive complexity being greater than 5 in the test files.

Reply:
This is fixed now: 18 violations in nested control flow, 6 in combination, 3 in recursion, 3 in breaks in linear flow. This is 30 total violations